internal/kafka/internal/services/kafka.go

```diff
@@ -400,20 +402,63 @@ func (k *kafkaService) RegisterKafkaJob(kafkaRequest *dbapi.KafkaRequest) *error
 		return errors.TooManyKafkaInstancesReached(fmt.Sprintf("region %s cannot accept instance type: %s at this moment", kafkaRequest.Region, kafkaRequest.InstanceType))
 	}
 
-	if !k.dataplaneClusterConfig.IsDataPlaneAutoScalingEnabled() {
-		cluster, e := k.clusterPlacementStrategy.FindCluster(kafkaRequest)
-		if e != nil || cluster == nil {
-			msg := fmt.Sprintf("no available cluster found for Kafka instance type '%s' in region '%s'", kafkaRequest.InstanceType, kafkaRequest.Region)
-			if e != nil {
-				logger.Logger.Error(fmt.Errorf("%s:%w", msg, e))
-			} else {
-				logger.Logger.Infof(msg)
-			}
+	// enterprise kafkas have their cluster ID assigned when a request is validated in the handler
+	if kafkaRequest.ClusterID != "" {
```

Comment:
I think this logic merits its place in the different placement strategies as it is one of the placement strategies.

I propose that this logic being moved onto https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/ba3f5f14f64443a12a20a2df6d38e045b3637054/internal/kafka/internal/services/cluster_placement_strategy.go 

and for each of the existing placement strategies (which are current scaling type dependent)

1. https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/ba3f5f14f64443a12a20a2df6d38e045b3637054/internal/kafka/internal/services/cluster_placement_strategy.go#L35
2. https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/ba3f5f14f64443a12a20a2df6d38e045b3637054/internal/kafka/internal/services/cluster_placement_strategy.go#L60
3. https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/ba3f5f14f64443a12a20a2df6d38e045b3637054/internal/kafka/internal/services/cluster_placement_strategy.go#L142

We've the check, "if the Kafka is enterprise (billing model is `enterprise` and cluster_id set)"  then apply this logic.
The advantage is that, the placement strategy is co-located in one place.